### PR TITLE
zippy: Use redpanda instead of kafka

### PR DESCRIPTION
**Pull request by def-**
For reduced memory consumption

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by nrainer-materialize — APPROVED** [review]
Consider triggering a nightly build with the zippy jobs.

**Comment by def-** [comment]
Running: https://buildkite.com/materialize/release-qualification/builds/410 and https://buildkite.com/materialize/nightlies/builds/6113

File: test/zippy/mzcompose.py [hunk 1]
```diff
@@ -40,9 +40,9 @@
 
 SERVICES = [
     Zookeeper(),
-    Kafka(auto_create_topics=True),
-    SchemaRegistry(),
-    Debezium(),
+    Redpanda(auto_create_topics=True),
+    SchemaRegistry(kafka_servers=[("redpanda", "9092")]),
```
Comment (philip-stoev) [inline test/zippy/mzcompose.py:44]: We need `redpanda=True` for `SchemaRegistry()` like we have for `Debezium`.

Comment (philip-stoev) [reply test/zippy/mzcompose.py:44]: Oh wait, no there is no need to run SchemaRegistry at all, it is included in the redpanda container.

File: test/zippy/mzcompose.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
     args = parser.parse_args()
     scenario_class = globals()[args.scenario]
 
-    c.up("zookeeper", "kafka", "schema-registry", "ssh-bastion-host")
+    c.up("zookeeper", "redpanda", "schema-registry", "ssh-bastion-host")
```
Comment (philip-stoev) [inline test/zippy/mzcompose.py:137]: No need for schema-registry at all here.

**Pull request merged**